File: PolishPizza/menu/src/main/java/pizzeria/menu/ingredient/service/IngredientService.java
package pizzeria.menu.ingredient.service;

import pizzeria.menu.ingredient.dto.response.IngredientResponse;
import pizzeria.menu.ingredient.dto.request.IngredientPatchRequest;
import pizzeria.menu.ingredient.dto.request.IngredientRequest;

import java.util.List;
import java.util.UUID;

public interface IngredientService {
    List<IngredientResponse> getAllIngredients();

    IngredientResponse getIngredientById(UUID id);

    IngredientResponse save(IngredientRequest request);

    void delete(UUID id);

    IngredientResponse update(UUID id, IngredientRequest request);

    IngredientResponse patch(UUID id, IngredientPatchRequest request);

}

File: PolishPizza/admin/src/service/ingredientService.ts
import { api as axiosInstance } from '../api/api';
import type { IngredientResponse, IngredientRequest } from '../types/menu';

export const IngredientService = {
    
    getAll: async (): Promise<IngredientResponse[]> => {
        const response = await axiosInstance.get<IngredientResponse[]>('/menu/ingredients'); 
        return response.data;
    },

    getOne: async (id: string): Promise<IngredientResponse | null> => {
        const response = await axiosInstance.get<IngredientResponse>(`/menu/ingredients/${id}`);
        return response.data;
    },
    
    update: async (id: string, data: IngredientRequest) => {
        const response = await axiosInstance.put(`/menu/ingredients/${id}`, data);
        return response.data;
    },
    
    create: async (data: IngredientRequest) => {
        const response = await axiosInstance.post(`/menu/ingredients`, data);
        return response.data;
    },

    delete: async(id: string): Promise<void> => {
        const response = await axiosInstance.delete(`/menu/ingredients/${id}`);
        return response.data;
    }
};